framework/post_processors/PostProcessorPrinter_00.cc

```diff
@@ -0,0 +1,205 @@
+#include "PostProcessorPrinter.h"
+#include "event_system/SystemWideEventPublisher.h"
+#include "event_system/EventSubscriber.h"
+#include "event_system/Event.h"
+
+#include "post_processors/PostProcessor.h"
+
+#include "chi_runtime.h"
+#include "chi_log.h"
+
+#include <algorithm>
+
+/**Small utility macro for joining two words.*/
+#define JoinWordsA(x, y) x##y
+/**IDK why this is needed. Seems like counter doesnt work properly without it*/
+#define JoinWordsB(x, y) JoinWordsA(x, y)
+
+std::shared_ptr<chi::PPPrinterSubscribeHelper>
+  chi::PostProcessorPrinter::helper_ptr_ =
+    std::make_shared<PPPrinterSubscribeHelper>(
+      PostProcessorPrinter::GetInstance());
+
+static char JoinWordsB(unique_var_name_ppp_, __COUNTER__) =
+  chi::PostProcessorPrinter::SubscribeToSystemWideEventPublisher();
+
+namespace chi
+{
+
+// ##################################################################
+PPPrinterSubscribeHelper::PPPrinterSubscribeHelper(
+  PostProcessorPrinter& printer_ref)
+  : printer_ref_(printer_ref)
+{
+}
+
+// ##################################################################
+/**Overrides base class to forward the call to the printer.*/
+void PPPrinterSubscribeHelper::ReceiveEventUpdate(const Event& event)
+{
+  printer_ref_.ReceiveEventUpdate(event);
+}
+
+// ##################################################################
+PostProcessorPrinter::PostProcessorPrinter()
+  : events_on_which_to_print_postprocs_({"SolverInitialized",
+                                         "SolverAdvanced",
+                                         "SolverExecuted",
+                                         "ProgramExecuted"})
+{
+}
+
+// ##################################################################
+PostProcessorPrinter& PostProcessorPrinter::GetInstance()
+{
+  static PostProcessorPrinter instance;
+
+  return instance;
+}
+
+// ##################################################################
+char PostProcessorPrinter::SubscribeToSystemWideEventPublisher()
```

Comment:
char? Maybe int?  I'm not sure why it's not void.

Reply:
Used at static init-time to subscribe to system-wide event handler.

At the top of the file:
```
static char JoinWordsB(unique_var_name_ppp_, __COUNTER__) =
  chi::PostProcessorPrinter::SubscribeToSystemWideEventPublisher();
```